### PR TITLE
#1028 (precursor) preferences endpoints

### PR DESCRIPTION
**Pull request by CarlosNZ**
In order to do #1028 properly, we need to have a UI to edit the preferences. This PR just provides a couple of additional Admin endpoints (`get-all-prefs`, `set-prefs`) to facilitate this.

Front-end PR https://github.com/openmsupply/conforma-web-app/pull/1497

Documentation for all preferences included here too.

### CODE REVIEW

File: src/components/files/createThumbnails.ts [hunk 1]
```diff
@@ -1,7 +1,11 @@
 import config from '../../config'
 import path from 'path'
 import sharp from 'sharp'
-const { thumbnailMaxWidth, thumbnailMaxHeight } = config
+import { DEFAULT_THUMBNAIL_MAX_WIDTH, DEFAULT_THUMBNAIL_MAX_HEIGHT } from '../../constants'
+const {
+  thumbnailMaxWidth = DEFAULT_THUMBNAIL_MAX_WIDTH,
+  thumbnailMaxHeight = DEFAULT_THUMBNAIL_MAX_HEIGHT,
+} = config
```
Comment (CarlosNZ) [inline src/components/files/createThumbnails.ts:8]: Tried to make as many prefs as possible have defaults.

File: src/components/permissions/routes.ts [hunk 2]
```diff
@@ -207,16 +203,6 @@ const routeVerification = async (request: any, reply: any) => {
...
-// Serve prefs to front-end
-const routeGetPrefs = async (request: any, reply: any) => {
-  const prefs = JSON.parse(
-    readFileSync(path.join(getAppEntryPointDir(), '../preferences/preferences.json'), 'utf8')
-  )
-  const languageOptions = readLanguageOptions()
-  const latestSnapshot = await databaseConnect.getLatestSnapshotName()
-  reply.send({ preferences: prefs.web, languageOptions, latestSnapshot })
-}
-
```
Comment (CarlosNZ) [inline src/components/permissions/routes.ts:219]: Just moved this to be with the new endpoints, as it's more related to them.

File: src/components/snapshots/useSnapshot.ts [hunk 3]
```diff
@@ -286,6 +286,7 @@ const copyFiles = async (snapshotFolder: string, fileRecords: ObjectRecord[] = [
 // Mutate the active config object to inject new preferences
 type Config = typeof config
 export const refreshPreferences = (config: Config) => {
+  console.log('Refreshing configuration...')
```
Comment (CarlosNZ) [inline src/components/snapshots/useSnapshot.ts:289]: So we know when the preferences should have been refreshed

File: documentation/Preferences.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+- **`previewDocsCleanupSchedule`**: The schedule for cleaning up (deleting) the preview documents, in node-schedule "hours" format, as above. (Default: once per day at 1 am)
+- **`backupSchedule`**: How often system backups should run, as per the node-schedule "hours" schedule above. (Default: once per day at 1 am)
+- **`backupFilePrefix`**: System backups are saved with the name format `backupFilePrefix_date_time.zip`, e.g. `conforma_backup_2023-04-04_01-00-00.zip`. (Default: "conforma_backup")
+- **`maxBackupDurationDays`**: Backups are kept for this many days, after which they're deleted next time the backup schedule runs. The default is nothing -- all backups will be kept.
+- **`testingEmail`**: During development and on a testing server, we don't want emails being sent to real people. If this property is set, and the site is not running on the designated host (as defined in `siteHost` below), then all emails will be send to this address instead. (If no `testingEmail` is specified, no emails will be sent at all)
```
Comment (CarlosNZ) [inline documentation/Preferences.md:30]: This property not actually implemented in this PR. Next one...

**Review by fergie-nz — APPROVED** [review]
This is awesome. I love this feature.

**Pull request merged**